test/e2e/tests/request-queuing/switchChain-watchAsset.spec.js

```diff
@@ -32,65 +32,28 @@ describe('Request Queue SwitchChain -> WatchAsset', function () {
             },
           ],
         },
+        smartContract,
         title: this.test.fullTitle(),
       },
 
-      async ({ driver }) => {
-        await unlockWallet(driver);
-
-        await openDapp(driver, undefined, DAPP_URL);
-
-        // Connect to dapp
-        await driver.findClickableElement({ text: 'Connect', tag: 'button' });
-        await driver.clickElement('#connectButton');
-
-        await driver.delay(regularDelayMs);
-
-        await switchToNotificationWindow(driver);
-
-        await driver.clickElement({
-          text: 'Next',
-          tag: 'button',
-          css: '[data-testid="page-container-footer-next"]',
-        });
-
-        await driver.clickElement({
-          text: 'Confirm',
-          tag: 'button',
-          css: '[data-testid="page-container-footer-next"]',
-        });
-
-        // Wait for Connecting notification to close.
-        await driver.waitUntilXWindowHandles(2);
-
-        // Navigate to test dapp
-        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
-
-        // Create Token
-        await driver.clickElement({ text: 'Create Token', tag: 'button' });
-        await switchToNotificationWindow(driver);
-        await driver.findClickableElement({ text: 'Confirm', tag: 'button' });
-        await driver.clickElement({ text: 'Confirm', tag: 'button' });
+      async ({ driver, contractRegistry, ganacheServer }) => {
+        const contractAddress = await contractRegistry.getContractAddress(
+          smartContract,
+        );
+        await logInWithBalanceValidation(driver, ganacheServer);
 
-        // Wait for token address to populate in dapp
-        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
-        await driver.wait(async () => {
-          const tokenAddressesElement = await driver.findElement(
-            '#tokenAddresses',
-          );
-          const tokenAddresses = await tokenAddressesElement.getText();
-          return tokenAddresses !== '';
-        }, 10000);
+        await openDapp(driver, contractAddress);
 
         // Switch Ethereum Chain
-        await driver.findClickableElement('#switchEthereumChain');
         await driver.clickElement('#switchEthereumChain');
 
         await driver.waitUntilXWindowHandles(3);
 
-        // Delay needed here for firefox, not loading page of switchEthereumChain before clicking the watchAsset button
-        // will result in watchAsset being front ran in the window.
-        await driver.delay(largeDelayMs);
```

Comment:
we can replace this delay, with the condition to be met before proceeding: finding the Network Switch screen in the popup